pandas/core/arrays/datetimelike.py

```diff
@@ -553,6 +553,23 @@ def shift(self, periods, freq=None):
         --------
         Index.shift : Shift values of Index.
         """
+        return self._tshift(periods=periods, freq=freq)
+
+    def _tshift(self, periods, freq=None):
```

Comment:
why don't we call this ._time_shift or ._period_shift

Reply:
if you are actually going to change this, then this should be much more consistent across the code base, e.g. does internals need changing, what about the Index classes?

Reply:
we have a note to deprecate tshift entirely. I don't think this is a good name at all.